commands/scanandfixrepos.go

```diff
@@ -33,12 +36,26 @@ func (saf *ScanAndFixRepositories) Run(configAggregator utils.FrogbotConfigAggre
 
 func (saf *ScanAndFixRepositories) scanAndFixSingleRepository(repoConfig *utils.FrogbotRepoConfig, client vcsclient.VcsClient) error {
 	for _, branch := range repoConfig.Branches {
-		err := saf.downloadAndRunScanAndFix(repoConfig, branch, client)
+		shouldScan, checkedCommit, err := saf.shouldScanLatestCommit(context.Background(), repoConfig, client, branch)
+		if err != nil {
+			return err
+		}
+		if !shouldScan {
+			log.Info(fmt.Sprintf("Commit %s has already been scanned by FrogBot", checkedCommit))
+			continue
+		}
+		err = saf.downloadAndRunScanAndFix(repoConfig, branch, client)
+		if err != nil {
+			errorDescription := fmt.Sprintf("FrogBot error %s", err)
+			log.Error(errorDescription)
```

Comment:
When a function an error, we usually avoid adding an error log message. This is beucase the error message will be added to the log anyway upstream.